### PR TITLE
[io-kafka] Cleanup offset committer references.

### PR DESCRIPTION
**Pull request by dmvk**

### CODE REVIEW

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java [hunk 1]
```diff
@@ -224,18 +224,20 @@ ObserveHandle processConsumer(
...
-                  if (commitToKafka) {
-                    kafkaCommitMap.put(tp, mtd);
-                  }
-                });
+    final BiConsumer<TopicPartition, ConsumerRecord<Object, Object>> preWrite =
+        (tp, r) -> {
+          final long offset = r.offset();
+          offsetCommitter.register(
+              tp,
+              r.offset(),
```
Comment (je-ik) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:232]: Can we make use of the new local variable here as well?

Comment (LesTR) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:232]: just offset?

Comment (LesTR) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:232]: ehm, next time I should refresh page before sending comment ....

Comment (dmvk) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:232]: the purpose of this change is that the inner lambda does not hold reference to ConsumerRecord (which can be potentially large as it contains serialized value + other metadata). This way we only hold reference for a single unboxed long value.

Comment (dmvk) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:232]: We can, but it's just a cosmetic change with no effect.

**Comment by je-ik** [comment]
LGTM :+1:

**Pull request merged**